content/cloud-docs/agents/hooks.mdx

```diff
@@ -0,0 +1,157 @@
+dynamically download software required by the Terraform run or send an HTTP
+request to a system to kick off an external workflow.
+
+## Supported Hooks
+
+The following hooks are supported:
+
+- `pre-plan` - The hook runs before `terraform init` during a `plan` operation.
+
+- `post-plan` - This hook runs after `terraform plan` during a `plan`
+operation regardless whether or not `terraform plan` completed successfully.
+
+- `pre-apply` - The hook runs before `terraform init` during an `apply`
+operation.
+
+- `post-apply` - This hook runs after `terraform apply` during an `apply`
+operation regardless whether or not `terraform apply` completed successfully.
+
+## Hook Behavior 
+
+Please note the following behavior when using hooks:
+
+- If a hook exits with a non-zero exit code, the Terraform run will fail
+immediately.
+- The standard output and standard error from the hook will be printed
+alongside the Terraform run output.
+- The name of each hook is not customizable.
+- Hooks must be executable. That is, each hook must have execute permission.
+- Each hook has a 60 second timeout.
+- There can only be one of each hook configured on an agent. That is, you can
+have both a `pre-plan` and `pre-apply` hook but you cannot have two `pre-plan`
+hooks.
+- Environment variables do not persist across hooks. For example, if a
+`pre-plan` hook exports environment variables, they will not be available
+during `terraform plan` nor will they be available during the `post-plan` hook.
+Similarly, if `terraform plan` exports environment variables, they will not be
+available during the `post-plan` hook.
+
+## Configuration
+
+Hooks are stored within the `hooks` subdirectory of the agent's data directory.
+By default, that's `~/.tfc-agent/hooks`.
+
+To configure a hook, create a file named `terraform-${HOOK}` where `${HOOK}`
+is one of the hooks listed above. Then store that file within the `hooks`
+subdirectory and make it executable. The agent will run your configured hooks
+as it processes Terraform runs.
+
+For example, a `pre-plan` hook is an executable file named
+`~/.tfc-agent/hooks/terraform-pre-plan`. When this hook is configured, an agent
+will run this hook before `terraform init` during a `plan` operation.
+
+### Running an Agent as a Binary
+
+Here's how you can configure hooks when running the `tfc-agent` binary
+directly on a Linux instance.
+
+1. Create the `hooks` directory within the agent's data directory.
+
+  ```sh
+  mkdir -p ~/.tfc-agent/hooks
+  ```
+
+1. Create your hooks. The following example creates a `pre-plan` hook.
+
+  ```sh
+  cat <<EOF > ~/.tfc-agent/hooks/terraform-pre-plan
+  #!/bin/bash
+  echo "Example pre-plan hook."
+  EOF
+  ```
+
+1. Make the hooks executable.
+
+  ```sh
+  chmod +x ~/.tfc-agent/hooks/terraform-pre-plan
+  ```
+
+1. Start the agent.
+
+  ```sh
+  tfc-agent
+  ```
+
+1. As the agent processes Terraform runs, it will run the hooks you've
+configured above.
```

Comment:
Also one thing that occurred to me is....do you think it could be helpful for users to see an example of what the output might look like? I know we say it prints "with the Terraform" stuff, but like where would they see it? Feel free to push back. but just thinking out loud. 

```suggestion
The agent will now trigger configured hooks at the appropriate points during Terraform runs.
```